src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs

```diff
@@ -347,7 +346,7 @@ private void RewriteTypeNames(TypeNameRewriter rewriter, ComponentIntermediateNo
                     attribute.GloballyQualifiedTypeName = globallyQualifiedTypeName;
                 }
 
-                if (attribute.BoundAttribute?.IsGenericTypedProperty() ?? false && attribute.TypeName != null)
+                if ((attribute.BoundAttribute?.IsGenericTypedProperty() ?? false) && attribute.TypeName != null)
```

Comment:
I'm a bit uncomfortable with this change without a test showing the impact, though it's pretty clear this is a bug. @chsienki, what do you think?

Reply:
Agree it would be nice to have a test but analyzing the code I think this is safe to make, as its just enforcing an assertion that is made elsewhere in the code. 

Line 370 will throw if the `TypeName` is null, so we can ignore that branch. Line 366 adds an annotation which is later checked for and the `TypeName` looked up which will throw if it's null. The only issue could arise if `globallyQualifiedTypeName` is not null, but `TypeName` is. In that case we'd set the `TypeName` to something when it isn't supposed to a value (because we missed this check) which could cause later issues. 

However, if we look at the implementation of `GetGloballyQualifiedTypeName` we can see that the attribute it's returning is only set in one place, where the `TypeName` is also set. That means the above can never be the case, and we're safe to expand the check (although as written in the codebase it's never strictly needed, hence why this code works correctly today anyway)